internal/pkg/agent/application/coordinator/coordinator.go

```diff
@@ -977,6 +999,8 @@ func (c *Coordinator) runLoopIteration(ctx context.Context) {
 	case upgradeDetails := <-c.upgradeDetailsChan:
 		c.setUpgradeDetails(upgradeDetails)
 
+	case c.heartbeatChan <- struct{}{}:
```

Comment:
question on this.  This looks like we will put something on the channel as soon as possible, so if the coordinator gets blocked after that, you would read from the heartbeatChan and think that it was up, because in the past it had been able to write to the channel.  On your next read it would fail.  Is that correct?

If it is, I'd rather see something that records a timestamp every time `runLoopIteration` is called, then we can check to see if that timestamp is within our timeout window.

Reply:
> so if the coordinator gets blocked after that, you would read from the heartbeatChan and think that it was up, because in the past it had been able to write to the channel. On your next read it would fail. Is that correct?

So, if I understand you correctly, yes. We could end up in a state where the coordinator blocks right after a heartbeat call. Because the liveness endpoint is meant to be repeated on some kind of regular period, I'm not too worried about that.

I specially didn't go with some kind of timestamp mechanism on @faec 's advice, since she was worried about the added complexity/edge cases of a time comparison, compared to a simple signal like this.

Reply:
I think this just tells us that the `runLoopIteration` function can run and the select can hit the heartbeat case.  It doesn't tell us if the coordinator can/has processed any of the other cases. 

I was thinking to be "alive", we want to know that one of the case statements besides heartbeat statement has run.  We would probably need to bound that comparison with the check-in Interval.  Or to put another way, the `runLoopIteration` function should happen at least every check-in Interval (2x is probably safer) and it might happen more frequently.

Reply:
Yeah, I sort of agree, this only really works as a basic heartbeat. 

My concern is that "has the coordinator done anything else?" would be a bit flaky, as we can't really guarantee what state other sub-components will be in at any given time. @faec can comment more, but for that to work we might need to refactor part of the coordinator to allow for a more sophisticated health check.